pkg/network/driver/nmstate/types.go

```diff
@@ -0,0 +1,234 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package nmstate
+
+import (
+	"net"
+
+	vishnetlink "github.com/vishvananda/netlink"
+
+	"kubevirt.io/kubevirt/pkg/network/driver/ethtool"
+	"kubevirt.io/kubevirt/pkg/network/driver/netlink"
+	"kubevirt.io/kubevirt/pkg/network/driver/procsys"
+	"kubevirt.io/kubevirt/pkg/network/driver/virtchroot"
+)
+
+type Spec struct {
+	Interfaces []Interface `json:"interfaces,omitempty"`
+	LinuxStack LinuxStack  `json:"linux-stack,omitempty"`
+}
+
+type Status struct {
+	Interfaces []Interface `json:"interfaces"`
+	Routes     Routes      `json:"routes"`
+	LinuxStack LinuxStack  `json:"linux-stack,omitempty"`
+}
+
+type Interface struct {
+	Name        string  `json:"name"`
+	Index       int     `json:"index,omitempty"`
+	TypeName    string  `json:"type,omitempty"`
+	State       string  `json:"state,omitempty"`
+	MacAddress  string  `json:"mac-address,omitempty"`
+	CopyMacFrom string  `json:"copy-mac-from,omitempty"`
+	MTU         int     `json:"mtu,omitempty"`
+	Controller  string  `json:"controller,omitempty"`
+	Ethtool     Ethtool `json:"ethtool,omitempty"`
+
+	Tap *TapDevice `json:"tap,omitempty"`
+
+	IPv4 IP `json:"IPv4,omitempty"`
+	IPv6 IP `json:"IPv6,omitempty"`
+
+	LinuxStack LinuxIfaceStack `json:"linux-stack,omitempty"`
+
+	Metadata *IfaceMetadata
+}
+
+type IP struct {
+	Enabled *bool       `json:"enabled,omitempty"`
+	Address []IPAddress `json:"address,omitempty"`
+}
+
+type IPAddress struct {
+	IP        string `json:"IP,omitempty"`
+	PrefixLen int    `json:"prefix-length,omitempty"`
+}
+
+type Routes struct {
+	Running []Route `json:"running,omitempty"`
+}
+
+type Route struct {
+	Destination      string `json:"destination"`
+	NextHopInterface string `json:"next-hop-interface,omitempty"`
+	NextHopAddress   string `json:"next-hop-address,omitempty"`
+	TableID          int    `json:"table-id,omitempty"`
+}
+
+type Ethtool struct {
+	Feature Feature `json:"feature,omitempty"`
+}
+
+type Feature struct {
+	TxChecksum *bool `json:"tx-checksum,omitempty"`
+}
+
+type TapDevice struct {
+	Queues int `json:"queues,omitempty"`
+	UID    int `json:"UID,omitempty"`
+	GID    int `json:"GID,omitempty"`
+}
+
+type LinuxIfaceStack struct {
```

Comment:
Why those two are inside `LinuxIfaceStack` and not flat like the other `Interface` properties?